### PR TITLE
DefinedSymbolCollector - Filter out anonymous classes.

### PR DESCRIPTION
**Pull request by SpacePossum**
closes https://github.com/maglnet/ComposerRequireChecker/issues/23

### CODE REVIEW

File: src/ComposerRequireChecker/NodeVisitor/DefinedSymbolCollector.php [hunk 1]
```diff
@@ -48,7 +48,7 @@ public function enterNode(Node $node)
 
     private function recordClassDefinition(Node $node)
     {
-        if ($node instanceof Node\Stmt\Class_) {
+        if ($node instanceof Node\Stmt\Class_ && !$node->isAnonymous()) {
```
Comment (Ocramius) [inline src/ComposerRequireChecker/NodeVisitor/DefinedSymbolCollector.php]: Is this API available in all versions of PHP-Parser? Just making sure we don't support older versions that don't have this.

Comment (SpacePossum) [inline src/ComposerRequireChecker/NodeVisitor/DefinedSymbolCollector.php]: I've tweaked the Travis config to test the lowest stable versions of the deps. on PHP7 and run the default on PHP 7.1. As to cover this case and future ones as well :)

File: .travis.yml [hunk 2]
```diff
@@ -1,29 +1,34 @@
...
+    depth: 1
 
 matrix:
-  fast_finish: true
-  allow_failures:
-    - php: nightly
+    fast_finish: true
+    include:
+        - php: 7.0
+          env: COMPOSER_FLAGS="--prefer-stable --prefer-lowest"
```
Comment (keradus) [inline .travis.yml]: where did I saw it ;P

Comment (SpacePossum) [inline .travis.yml]: yeah guess what I tried before this line : P
I still consider this a compoesr ~bug~ DX issue

File: .travis.yml [hunk 3]
```diff
@@ -1,29 +1,34 @@
...
-  - sed s\@TRAVIS_BUILD_DIR@$TRAVIS_BUILD_DIR@g test/travis-composer.json > ~/.composer/composer.json
+    - composer self-update
+    - sed s\@TRAVIS_BUILD_DIR@$TRAVIS_BUILD_DIR@g test/travis-composer.json > ~/.composer/composer.json
 
 install:
-  - composer install -vvv
-
-  - composer global require maglnet/composer-require-checker:@dev
-  - export PATH=$PATH:~/.composer/vendor/bin
+    - composer update $COMPOSER_FLAGS --no-interaction --no-progress --optimize-autoloader -vvv
```
Comment (SpacePossum) [inline .travis.yml]: thanks, done :+1:

**Pull request merged**